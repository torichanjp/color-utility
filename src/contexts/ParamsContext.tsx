import React, {useContext, useState} from "react";

export type ParamsType = {
    hueDivisions: number,
    setHueDivisions: Function,
    useLuma: boolean,
    setUseLuma: Function,
}

const defaultValues = {
    hueDivisions: 10,
    setHueDivisions: () => {
        console.warn('ParamsContext uses default value')
    },
    useLuma: true,
    setUseLuma: () => {
        console.warn('ParamsContext uses default value')
    },
}

const Params = React.createContext<ParamsType>(defaultValues)
export const useParams = () => useContext(Params)

export const ParamsContext: React.FC<{children: React.ReactNode}> = ({children}) => {
    const [hueDivisions, setHueDivisions] = useState(10)
    const [useLuma, setUseLuma] = useState(true)

    const values = {
        hueDivisions,
        useLuma,
        setHueDivisions,
        setUseLuma,
    }

    return (
        <Params.Provider value={values}>
            {children}
        </Params.Provider>
    )
}
