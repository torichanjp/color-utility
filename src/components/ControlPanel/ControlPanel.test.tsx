import React from 'react';
import { render, screen } from '@testing-library/react';
import { ControlPanel } from './index';
import userEvent from "@testing-library/user-event";
import {ParamsContext} from "../../contexts/ParamsContext";

test('ControlPanel has 4 buttons', async () => {
    render(<ControlPanel />);
    const buttons = await screen.queryAllByRole('button');
    expect(buttons).toHaveLength(4);
});

test('ControlPanel has a textbox', async () => {
    render(<ControlPanel />);
    const textBox = await screen.queryByRole('textbox');
    expect(textBox).not.toBeNull();
});

const buttonTest = async (name: string) => {
    const value = parseInt(name)

    render(<ParamsContext><ControlPanel /></ParamsContext>);
    const button = screen.getByRole("button", { name })
    const textBox = screen.getByLabelText("num-of-hue-divisions") as HTMLInputElement
    const currentValue = parseInt(textBox.value || "-99")
    await userEvent.click(button)
    expect(textBox.value).toBe((currentValue + value).toString())
}
test("-1 button affects the textbox.", async () => {
    await buttonTest("-1")
});

test("-10 button affects the textbox.", async () => {
    await buttonTest("-10")
});

test("+1 button affects the textbox.", async () => {
    await buttonTest("+1")
});

test("+10 button affects the textbox.", async () => {
    await buttonTest("+10")
});

test("Snap shot for ControlPanel", () => {
    const view = render(<ControlPanel />);
    expect(view.container).toMatchSnapshot();
});
