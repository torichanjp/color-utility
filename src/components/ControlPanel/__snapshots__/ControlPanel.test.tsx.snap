// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snap shot for ControlPanel 1`] = `
<div>
  <div
    class="wrapper"
  >
    <div
      class="firstSortKey"
    >
      <p>
        First sort key: HSV Hue
      </p>
      <p
        class="description"
      >
        How many divisions of HSV Hue?
      </p>
    </div>
    <button>
      -10
    </button>
    <button>
      -1
    </button>
    <input
      aria-label="num-of-hue-divisions"
      type="text"
      value="10"
    />
    <button>
      +1
    </button>
    <button>
      +10
    </button>
    <div
      class="secondSortKey"
    >
      <p>
        Second sort key
      </p>
      <input
        checked=""
        id="luma"
        type="radio"
        value="luma"
      />
      <label
        for="luma"
      >
        Luma
      </label>
      <input
        id="hsv"
        type="radio"
        value="hsv"
      />
      <label
        for="hsv"
      >
        HSV Value
      </label>
    </div>
  </div>
</div>
`;
